utils/create_dataset.py

```diff
@@ -104,3 +114,155 @@ def __getitem__(self, index):
                                 f"\nIgnore if some augmentations have padded with dontcare value.")
         sample['index'] = index
         return sample
+
+
+class VRTDataset(GeoDataset):
+    def __init__(self, vrt_ds: DatasetReader) -> None:
+        """Initialize a new VRTDataset instance.
+        The dataset is base on the DataReader class, initiated by rasterio.open().
+
+        Args:
+            vrt_ds: DatasetReader object (rasterio)
+        """
+        super().__init__()
+
+        self.vrt_ds = vrt_ds
+        try:
+            self.cmap = vrt_ds.colormap(1)
+        except ValueError:
+            pass
+
+        crs = vrt_ds.crs
+        res = vrt_ds.res[0]
+
+        with WarpedVRT(vrt_ds, crs=crs) as vrt:
+            minx, miny, maxx, maxy = vrt.bounds
+
+        mint: float = 0
+        maxt: float = sys.maxsize
+
+        coords = (minx, maxx, miny, maxy, mint, maxt)
+        self.index.insert(0, coords, 'vrt')
+
+        self._crs = cast(CRS, crs)
+        self.res = cast(float, res)
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image and metadata at that index
+        """
+        data = self._get_tensor(query)
+        key = "image"
+        sample = {key: data, "crs": self.crs, "bbox": query}
+
+        return sample
+
+    def _get_tensor(self, query):
+        """
+        Get a patch based on the given query (bounding box).
+        Args:
+            query:
+
+        Returns: Torch tensor patch.
+
+        """
+        bounds = (query.minx, query.miny, query.maxx, query.maxy)
+        out_width = round((query.maxx - query.minx) / self.res)
+        out_height = round((query.maxy - query.miny) / self.res)
+        out_shape = (self.vrt_ds.count, out_height, out_width)
+
+        dest = self.vrt_ds.read(
+            out_shape=out_shape, window=from_bounds(*bounds, self.vrt_ds.transform)
+        )
+
+        if dest.dtype == np.uint16:
+            dest = dest.astype(np.int32)
+        elif dest.dtype == np.uint32:
+            dest = dest.astype(np.int64)
+
+        tensor = torch.tensor(dest)
+
+        return tensor
+
+
+class GDLVectorDataset(GeoDataset):
+    """Abstract base class for :class:`GeoDataset` stored as vector files."""
```

Comment:
This may have been copied from torchgeo, but doesn't seem accurate here. It's not an abstract base class.

Also, you previously used the VectorDataset, why go back to the more generic GeoDataset? Was VectorDataset not applicable for our use case? Same question holds for switch from RasterDataset to GeoDataset above.

Reply:
Changed the class docstring.